### ISSUE TITLE
[Bug Report][3.1.6] Menu wrongly positioned in dialog after scrolling (only when first open)

### ISSUE THREAD

**Issue by JosuaMeier** [vuetifyjs/vuetify#16770]
### Environment
**Vuetify Version:** 3.1.6
**Last working version:** 3.1.2
**Vue Version:** 3.2.47
**Browsers:** Chrome 110.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
* Open reproduction link
* Scroll down a bit
* Open dialog via button
* Open the select menu

### Expected Behavior
Menu should be directly under the select input. On the second attempt, the menu is correctly positioned.

### Actual Behavior
Menu is not visible because its not directly under the menu.

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqVUs1O8zAQfBXL34FLk/TjVwo/gifgwq3pwU2W1JDYK3sbQFXfnbUTSpMKJC5RdnYyMzvKYisfENNuAzKXNwQtNorgrjBC3HSJQoyvcWiVNsMU59IaYgjcHoxwpVVj6wOsJytXjbCIemigJJFrNva3hVycPIGnk2Uhp9zsizwWZvhIeX+F+KdK0p0i61h7i86i3x1Ld8mKjOCnNhXzIo1Zjwgm6PNy4jlq6TvI5PCYbVJRwPY1hqEveCTJoy+dRhKNMjUHIk4jPNAmUAO5Xw9U+mhA+NIihBbSA0+xDS5r0PWacnE2n8/x/TpAqKpKmzohi7m4GOBdVA5qd3ImdYvWEWfF9MVbw/9GFCuGBSfKe/mA8c8T5kKuidDnWbYx+FqnpW2ze95lbmNIt5BUtr0/S0/T86us0p4O8RR8m6ycffPg2LGQswNx0s8fPxkMa9b9n172sgOUNGrlg+6RXsYmHbjEganAgfs1/IQ7OmCyOzoieHKvOy600ebVT7osfexx8RXt79dFhfD5MrjsZrJPx37xC7n8BH/VOqE=)

### Other comments
Calculation of the top attribute is not correct.

**Comment by benoitkugler** [comment]
This is probably a duplicate of #16663

**Comment by qxygene** [comment]
any fix yet?

**Comment by naranjamecanica** [comment]
Maybe of help: when I scroll the page menu suddenly appears.

**Comment by qxygene** [comment]
issue with [scroll-strategy](https://vuetifyjs.com/en/api/v-dialog/#props-scroll-strategy) 'block' is problem.

**Comment by avarixe** [comment]
My solution to this was to set `overflow: hidden` to `<html>` when the dialog is opened. That way, it still acts similarly enough to scroll-strategy: block:

```js
const dialog = ref(false)
watchEffect(() => {
  document.documentElement.style.overflow = dialog.value ? 'hidden' : null
})

....
<v-dialog v-model="dialog">
```

**Comment by joel-wenzel** [comment]
@KaelWD Did you already have an idea about how you wanted to fix this one?
The issue seems to be in `updateLocation` of connected location strategy:
```
contentBox.y += parseFloat(document.documentElement.style.getPropertyValue('--v-body-scroll-y') || 0);
```
contentBox.y is negative and so is v-body-scroll. So its actually doubling the offset vs correcting it back to 0. Just changing to `-=` fixes the problem

**Comment by qxygene** [comment]
> A simple workaround to this issue is adding `scroll-strategy="none"` to v-dialog.
> 
> ```
> <v-dialog
>     v-model="dialog"
>     scroll-strategy="none"
> >
> ```
> 
> This has the drawback that the underlying page is scrollable when the dialog is open. You'll see the content scrolling under the dialog and the overlay.

Yes this works, but when you go to xs with inner scrollbar(overflow-y-auto) that will give you 2 scrollbars side by side.  Needs a proper fix.

**Comment by WhiteSage** [comment]
Any progress on this issue? I've recently been forced to update to 3.1.5 (version 3.1.2 seems to be the last one where this was working correctly) and using some of the dialogs has become quite painful.

**Comment by gdgcentrica** [comment]
This problem is not resolved in v3.2.2

**Comment by KaelWD** [comment]
Correct

**Comment by JosuaMeier** [comment]
@gdgcentrica Should be resolved in 3.2.3.
